providers/standard/src/airflow/providers/standard/triggers/file.py

```diff
@@ -75,3 +86,50 @@ async def run(self) -> typing.AsyncIterator[TriggerEvent]:
                         yield TriggerEvent(True)
                         return
             await asyncio.sleep(self.poke_interval)
+
+
+class FileDeleteTrigger(BaseEventTrigger):
+    """
+    A trigger that fires exactly once after it finds the requested file and then delete the file.
+
+    The difference between ``FileTrigger`` and ``FileDeleteTrigger`` is ``FileDeleteTrigger`` can only find a
+    specific file. When this file is found out, ``FileDeleteTrigger`` return the content of this file after
+    deleting it.
+
+    :param filepath: File (relative to the base path set within the connection).
+    :param poke_interval: Time that the job should wait in between each try
+    """
+
+    def __init__(
+        self,
+        filepath: str,
+        poke_interval: float = 5.0,
+        **kwargs,
+    ):
+        super().__init__()
+        self.filepath = filepath
+        self.poke_interval = poke_interval
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serialize FileDeleteTrigger arguments and classpath."""
+        return (
+            "airflow.providers.standard.triggers.file.FileDeleteTrigger",
+            {
+                "filepath": self.filepath,
+                "poke_interval": self.poke_interval,
+            },
+        )
+
+    async def run(self) -> typing.AsyncIterator[TriggerEvent]:
```

Comment:
Nit: we should be using the `collections.abc` version of this, not the typing one

Reply:
Good catch!